### PR TITLE
[wgsl] Alignment attribute tests.

### LINKED ISSUE #1432: wgsl: Validate `@align` parsing
# On a struct member
* [x] `@malign(1)` -- misspelling
* [x] `@align()` -- missing value
* [x] `@align 1)`  -- missing (
* [x] `@align(1 ` -- missing )
* [x] `@align(1, 2)` -- multiple values
* [x] `@align(identifier)` -- must be integer literal
* [x] `@align(1.0)` -- must be integer literal
* [x] `@align(-1)` -- non-negative
* [x] `@align(3)` -- must be a multiple of 2
* [x] `@align(1)` -- valid
* [x] `@align(1i)` -- valid
* [x] `@align(1u)` -- invalid
* [x] `@align(1f)` -- invalid
* [x] `@align(1h)` -- invalid
* [x] `@align(1,)` -- valid trailing comma
* [x] `@align(0x10)` -- (validate hex is accepted)
* [x] `@align(2147483648)` -- valid?
* [x] `@align`
* [x] `@ \t align`
* [x] `@/* comment */align`
* [x] `@align(0)` - invalid must be positive

# Non-struct member tests
* [x] For each type, try to align without satisfying the required-alignment for the type
* [ ] Add align to non-struct type, should reject
   * [ ] align on vector
   * [ ] align on matrix
   * [ ] align of struct itself instead of member
   * [ ] align of array
   * [ ] align on const
   * [ ] align on scalar

# Misc
* [ ] Validate `@align` can only be provided once on a given member.

### PR DESCRIPTION
**Pull request by dj2**
This CL adds test for alignment attribute used outside as struct and multiple alignment attributes applied to an item.

Fixes #1432

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by dj2** [comment]
I intially wrote `align_non_struct` with all of the possible types in there. The test got pretty unreadable when starting to deal with things that have to be in `read_write` storage, or can't be in `storage` at all. It seemed a lot simpler just to drop back and check a single case but we'll have to decide if we want to expand that.

**Review by zoddicus — COMMENTED** [review]
This looks reasonable to me, but I would like @ben-clayton to weigh in on if this needs to be expanded, since he is more familiar with the spec alignment requirements.

File: src/webgpu/shader/validation/parse/align.spec.ts [hunk 1]
```diff
@@ -178,3 +178,62 @@ g.test('align_required_alignment')
 
     t.expectCompileResult(!fails, code);
   });
+
+g.test('align_non_struct')
+  .desc('Test that alignment is only allowed on struct members')
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/parse/align.spec.ts:183]: This doesn't seem very exhaustive in terms of where the attribute can go. 

Would it be better to parameterize a list of shaders, with insertion points for attributes, then have a big table of `attribute + location = valid` that we can check? That way, you can validate nearly all the attributes for all locations.

**Review by ben-clayton — APPROVED** [review]
Approving what you have, as a first step. As discussed, I think this would be better to parameterize.

File: src/webgpu/shader/validation/parse/align.spec.ts (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/parse/align.spec.ts]: Done. Made it specific to align at the moment. Want to get a few more in place to see if/how it generalizes.

**Pull request merged**